src/Context/SoapContext.php

```diff
@@ -83,6 +83,16 @@ public function iSendRequestWithParams($function, TableNode $params)
     {
         $this->sendRequest($function, [$params->getRowsHash()]);
     }
+    
+    /**
+     * Send SOAP request with params as nested array.
+     *
+     * @Given I call SOAP function :function with params as json:
+     */
+    public function iSendRequestWithNestedParamsArray($function, PyStringNode $json)
+    {
+        $this->sendRequest($function, [json_decode('{' . $json . '}', 'true')]);
```

Comment:
Anyway I just realized now that there is already an equivalent function but based on yml format instead of json. So I guess this pull request can be dropped.